src/rtc/core.py

```diff
@@ -152,6 +156,327 @@ def save_as_cog(filename, scratch_dir = '.', logger = None,
+               excluding the shoreline shapefile
+       check_shoreline_shapefile: bool
+               Flag that checks for the shoreline shapefile
+       dem_file: str
+               DEM filename
+       geogrid: isce3.product.GeoGridParameters
+               Product's ISCE3 geogrid object
+       metadata_dict: collections.OrderedDict
+               Metadata dictionary
+       logger : logging.Logger, optional
+              Logger object
+    """
+    if logger is None:
+        logger = logging.getLogger('rtc_s1')
+    logger.info(f"Check ancillary inputs' coverage:")
+
+    # file description (to be printed to the user if an error happens)
+    dem_file_description = 'DEM file'
+
+    if not check_ancillary_inputs_coverage:
+
+        # print messages to the user
+        logger.info(f'    {dem_file_description} coverage:'
+                    ' (not tested)')
+
+        # update RTC-S1 product metadata
+        metadata_dict['DEM_COVERAGE'] = 'NOT_TESTED'
+
+        return
+
+    rasters_to_check_dict = {'DEM': (dem_file_description, dem_file)}
+
+    geogrid_x0_projected = geogrid.start_x
+    geogrid_y0_projected = geogrid.start_y
+    geogrid_xf_projected = (geogrid.start_x +
+                            geogrid.spacing_x * geogrid.width)
+    geogrid_yf_projected = (geogrid.start_y +
+                            geogrid.spacing_y * geogrid.length)
+
+    geogrid_srs = osr.SpatialReference()
+    geogrid_srs.ImportFromEPSG(geogrid.epsg)
+
+    for ancillary_file_type, \
+        (ancillary_file_description, ancillary_file_name) in \
+            rasters_to_check_dict.items():
+
+        # check if file was provided
+        if not ancillary_file_name:
+            error_msg = f'ERROR {ancillary_file_description} not provided'
+            logger.error(error_msg)
+            raise ValueError(error_msg)
+
+        # check if file exists
+        if not os.path.isfile(ancillary_file_name):
+            error_msg = f'ERROR {ancillary_file_description} not found:'
+            error_msg += f' {ancillary_file_name}'
+            logger.error(error_msg)
+            raise FileNotFoundError(error_msg)
+
+        # test if the reference geogrid is fully covered by the ancillary input
+        # by checking all ancillary input vertices are located
+        # outside of the reference geogrid.
+        ancillary_gdal_ds = gdal.Open(ancillary_file_name, gdal.GA_ReadOnly)
+
+        ancillary_geotransform = ancillary_gdal_ds.GetGeoTransform()
+        ancillary_projection = ancillary_gdal_ds.GetProjection()
+        ancillary_x0, ancillary_dx, _, ancillary_y0, _, ancillary_dy = \
+            ancillary_geotransform
+        ancillary_width = ancillary_gdal_ds.GetRasterBand(1).XSize
+        ancillary_length = ancillary_gdal_ds.GetRasterBand(1).YSize
+
+        del ancillary_gdal_ds
+
+        ancillary_xf = ancillary_x0 + ancillary_width * ancillary_dx
+        ancillary_yf = ancillary_y0 + ancillary_length * ancillary_dy
```

Comment:
Not so clear to me about what `f` means in `ancillary_xf` and `ancillary_yf`. Same for `geogrid_xf` and `geogrid_yf`.

Reply:
They are the final coordinates `[x0:xf]`; `[y0:yf]`. I added some comments to make this clearer